app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editselfdeletingmessages/EditSelfDeletingMessagesViewModel.kt

```diff
@@ -0,0 +1,146 @@
+    private val navigationManager: NavigationManager,
+    private val dispatcher: DispatcherProvider,
+    private val observeConversationMembers: ObserveParticipantsForConversationUseCase,
+    private val observeSelfDeletionTimerSettingsForConversation: ObserveSelfDeletionTimerSettingsForConversationUseCase,
+    private val updateMessageTimer: UpdateMessageTimerUseCase,
+    savedStateHandle: SavedStateHandle,
+    qualifiedIdMapper: QualifiedIdMapper,
+) : ViewModel() {
+
+    val conversationId: QualifiedID = qualifiedIdMapper.fromStringToQualifiedID(
+        checkNotNull(savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)) {
+            "No conversationId was provided via savedStateHandle to EditSelfDeletingMessagesViewModel"
+        }
+    )
+
+    var editSelfDeletingMessagesState by mutableStateOf(
+        EditSelfDeletingMessagesState()
+    )
+
+    init {
+        observeSelfDeletingMessages()
+    }
+
+    private fun observeSelfDeletingMessages() {
+        viewModelScope.launch {
+            combine(
+                observeSelfDeletionTimerSettingsForConversation(conversationId, considerSelfUserSettings = false),
+                observeConversationMembers(conversationId).map { it.isSelfAnAdmin },
+                ::Pair
+            )
+                .distinctUntilChanged()
+                .flowOn(dispatcher.io())
+                .collect { (selfDeletingMessages, isSelfAnAdmin) ->
+                    editSelfDeletingMessagesState = editSelfDeletingMessagesState.copy(
+                        isLoading = selfDeletingMessages.isEnforcedByTeam || !isSelfAnAdmin,
+                        isEnabled = selfDeletingMessages.isEnforcedByGroup,
+                        selfDeletingDuration = selfDeletingMessages.toDuration(),
+                        currentlySelected = selfDeletingMessages.toDuration().toSelfDeletionDuration()
+                    )
+                }
+        }
+    }
+
+    fun updateSelfDeletingMessageOption(shouldBeEnabled: Boolean) {
+        viewModelScope.launch {
+            if (shouldBeEnabled) {
+                editSelfDeletingMessagesState = editSelfDeletingMessagesState.copy(
+                    isEnabled = true
+                )
+            } else {
+                editSelfDeletingMessagesState = editSelfDeletingMessagesState.copy(
+                    isLoading = true,
+                    isEnabled = false
+                )
+                editSelfDeletingMessagesState = when (updateMessageTimer(conversationId, null)) {
+                    is UpdateMessageTimerUseCase.Result.Failure -> editSelfDeletingMessagesState.copy(
+                        isLoading = false,
+                        isEnabled = true
+                    )
+
+                    UpdateMessageTimerUseCase.Result.Success -> editSelfDeletingMessagesState.copy(
+                        isLoading = false,
+                        isEnabled = false
+                    )
+                }
+            }
```

Comment:
I think it is possible to do it at one place i am talking about settings isLoading = false

Reply:
done